### ISSUE TITLE
Add option to one-var to allow for separating function expressions

### ISSUE THREAD

**Issue by andree-rehnberg** [eslint/eslint#14222]
**What rule do you want to change?**
one-var

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option, in the same way that one-var already has the option "separateRequires"

**Please provide some example code that this change will affect:**

```js
const variableOne = 'lorem',
  variableTwo = 'ipsum';

const functionExpression = () => {
  // This function block will get an error if one-var is set to "always" or "consecutive".
}
```

**What does the rule currently do for this code?**
With one-var set to `always` or `consecutive` the function expression will get an error telling you to combine it with the previous const statement.

**What will the rule do after it's changed?**
With an option named something like `separateFunctionExpressions` set to `true` function expressions should be allowed to be declared with it's own const statement instead of having to be combined with the previous one.

The reasoning for this is to make it clearer where the function expression starts and stops, this is mainly a problem when having many large functions placed after each other.

**Are you willing to submit a pull request to implement this change?**
Maybe

**Comment by mdjermanovic** [comment]
Hi @andree-rehnberg , thanks for the issue!

The team has made the difficult decision to freeze stylistic rules (please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)), so we won't be adding any new options to the `one-var` rule. You can always make a custom rule or publish a [plugin](https://eslint.org/docs/developer-guide/working-with-plugins).

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).
